team/views.py

```diff
@@ -0,0 +1,328 @@
+        return send_json_response('NOT_ALLOWED', 'Not Allowed', {**errors.get_dict_erros()})
+    
+    # check if Team object exists
+    if not Team.objects.filter(pk=team_id).exists():
+        errors.add(0, 'id', 'Team with id: {} not found'.format(team_id))
+        return send_json_response('NOT_FOUND', 'Not Found', {**errors.get_dict_erros()})
+
+    team = team_normalizer(Team.objects.get(pk=team_id))
+    return send_json_response('SUCCESS', 'success', team)
+
+
+def add_team(request):
+    """POST new team
+
+    Args:
+        request (class): Django request Class
+
+    Request body content: 
+        label (str): Its label.
+        description (str): Its description.
+
+    Returns:
+        dict: response with new team's informations
+    """
+    errors = Errors()
+    
+    # request method must be POST type
+    if not request_method_is(request, 'POST'):    
+        errors.add(0, 'method', 'Must be a POST method')
+        return send_json_response('NOT_ALLOWED', 'Not Allowed', {**errors.get_dict_erros()})
+
+    content = json.loads(request.body.decode('utf-8'))
+    
+    # check if Organizaton object exists
+    if not 'organization' in content or not Organization.objects.filter(id=content['organization']).exists():
+        errors.add(0, 'organization', 'Organization not found')
+        return send_json_response('INTERNAL_SERVER_ERROR', 'error', {**errors.get_dict_erros()})
+
+    content['organization'] = Organization.objects.get(id=content['organization'])    
+    form = TeamForm(content)
+
+    # if form is not valid, display form errors
+    if not form.is_valid():
+        # loop and append errors from erros.items()
+        for form_error in list(form.errors.items()):
+            errors.add(0, form_error[0], ', '.join(form_error[1]))
+        return send_json_response('INTERNAL_SERVER_ERROR', 'error', {**errors.get_dict_erros()})
+
+    # save and get team object
+    team_object = form.save()
+    team = team_normalizer(Team.objects.get(pk=team_object.id))
+    return send_json_response('SUCCESS', 'success', team)
+
+
+def update_team(request):
+    """PATCH team informations with the id in the body
+
+    Args:
+        request (class): Django request Class
+
+    Request body content: 
+        id (int): The id of team. Team object with this id must be exists.
+        label (str): Its label.
+        description (str): Its description.
+
+    Returns:
+        dict: response with new team's informations
+    """
+    errors = Errors()
+    
+    # request method must be PATCH type
+    if not request_method_is(request, 'PATCH'):    
+        errors.add(0, 'method', 'Must be a PATCH method')
+        return send_json_response('NOT_ALLOWED', 'Not Allowed', {**errors.get_dict_erros()})
+
+    content = json.loads(request.body.decode('utf-8'))
+    team_id = content['id']
+
+    # check if Team object exists
+    if not Team.objects.filter(pk=team_id).exists():
```

Comment:
Pourquoi ne pas favoriser les try/except plutôt que les if dans des cas où la requête est supposée toujours envoyer les bons ids mais qu'il peut arriver que parfois cet id soit mauvais ?

Le try/except est majoritairement plus rapide lorsqu'on envoie le bon id 99% du temps. (cf.: [try except vs if in python](https://www.geeksforgeeks.org/try-except-vs-if-in-python/) et [difference between EAFP and LBYL](https://stackoverflow.com/a/1835844))

liens supplémentaires:
[EAFP style](https://docs.python.org/3.6/glossary.html#term-eafp)
[LBYL style](https://docs.python.org/3.6/glossary.html#term-lbyl)